### PR TITLE
homework initial commit

### PR DESCRIPTION
**Pull request by hovhaik**

### CODE REVIEW

File: homework-20-12-23/hayk-hovhannisyan/vpc_create.sh [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+#add route to subnet
+associate_response=$(aws ec2 associate-route-table --subnet-id "$subnetId" --route-table-id "$routeTableId")
+echo " "
+echo "VPC created:"
+echo "Use subnet id $subnetId and security group id $groupId"
+echo "To create your AWS instances"
+# end of create-aws-vpc
+
+
+ami_id=ami-0fc5d935ebf8bc3bc
```
Comment (shtajiryan) [inline homework-20-12-23/hayk-hovhannisyan/vpc_create.sh]: ami-id should be the Free Tier Eligible Ubuntu. This is a community ami-id, which may or may not generate cost.

File: homework-20-12-23/hayk-hovhannisyan/vpc_create.sh [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+instanse_type=t2.micro
+key_pair=$$_key_vpc
+aws ec2 create-key-pair --key-name $key_pair --query 'KeyMaterial' --output text > $key_pair.pem
```
Comment (shtajiryan) [inline homework-20-12-23/hayk-hovhannisyan/vpc_create.sh]: generating a new key pair is not best practice, especially considering you might be running this script off of a GitHub Worker node, subsequently putting a Private SSH key on a machine owned by a third party.

File: homework-20-12-23/hayk-hovhannisyan/vpc_create.sh [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    --query 'Instances[0].InstanceId' \
+    --output text) 
+echo "Instance launched with id $instanse_id"
+
+instance_ip=$(aws ec2 describe-instances \
+    --instance-ids $instanse_id \
+    --query "Reservations[0].Instances[0].PublicIpAddress" --output text)  
+echo "EC2 instance myServer1 IP: $instance_ip"
+echo "To connect $instance_ip instance"
+echo "sudo ssh -i  $key_pair.pem ubuntu@$instance_ip"
```
Comment (shtajiryan) [inline homework-20-12-23/hayk-hovhannisyan/vpc_create.sh]: ssh should not be run with sudo

File: homework-20-12-23/hayk-hovhannisyan/vpc_create.sh [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+echo "to delete  key pair"
+echo "aws ec2 delete-key-pair --key-name $key_pair --region us-east-1"
```
Comment (shtajiryan) [inline homework-20-12-23/hayk-hovhannisyan/vpc_create.sh]: you can use your primary key pair, with the private key situated safely in your machine.

**Review by shtajiryan — CHANGES_REQUESTED** [review]
Creation script creates all resources successfully.
Creation script should have outputs indicating on which step of which resource creation the script is running at that moment, this enables other users to be able to troubleshoot on which step something fails.

cleanup script refuses to work with the following error:

`jq: error (at <stdin>:314): Cannot iterate over null (null)
Sleeping for 30 seconds while waiting for instances to be fully terminated.`

and continues to loop for 30 seconds, meanwhile the instance stays in a running state.

File: homework-20-12-23/hayk-hovhannisyan/vpc_create.sh (hunk 1 above)
Comment (Gev544) [reply homework-20-12-23/hayk-hovhannisyan/vpc_create.sh]: ami-0c7217cdde317cfec

**Comment by shtajiryan** [comment]
The scripts work correctly.

**Review by shtajiryan — APPROVED** [review]

**Pull request merged**